lib/azure/armrest/insights/event_service.rb

```diff
@@ -39,63 +44,35 @@ def initialize(armrest_configuration, options = {})
         #   filter = "eventTimestamp ge #{date} and eventChannels eq 'Admin, Operation'"
         #   select = "resourceGroupName, operationName"
         #
-        #   ies.list(filter, select, skip_token).events.each{ |event|
+        #   ies.list(:filter => filter, :select => select, :all => true).events.each{ |event|
         #     p event
         #   }
         #
-        def list(filter = nil, select = nil, skip_token = nil)
-          url = build_url(filter, select, skip_token)
+        def list(options = {})
+          url = build_url(options)
           response = rest_get(url)
           json_response = JSON.parse(response.body)
 
-          events          = json_response['value'].map{ |e| Azure::Armrest::Insights::Event.new(e) }
-          next_link       = json_response['nextLink']
-          next_skip_token = next_link[/.*?skipToken=(.*?)$/, 1] if next_link
-          Azure::Armrest::Insights::EventList.new(events, next_skip_token)
-        end
+          events = ArmrestCollection.new(json_response['value'].map { |e|
+            Azure::Armrest::Insights::Event.new(e)
+          })
 
-        # Returns a list of management events for the current subscription.
-        # The +filter+ option can be used to filter the result set. Additionally,
-        # you may restrict the results to only return certain fields using
-        # the +select+ option. The possible fields for both filtering and selection
-        # are:
-        #
-        # authorization, channels, claims, correlationId, description, eventDataId,
-        # eventName, eventSource, eventTimestamp, httpRequest, level, operationId,
-        # operationName, properties, resourceGroupName, resourceProviderName,
-        # resourceUri, status, submissionTimestamp, subStatus and subscriptionId.
-        #
-        # +list_all+ differs from the +list+ method in that it will list all of
-        # the events that would be returned for the given +filter+ and +select+
-        # instead of returning only the first page of events as defined by the
-        # Azure API event threshold (200 per page, by default).
-        #
-        # Example:
-        #
-        #   ies = Azure::Armrest::Insights::EventService.new(conf)
-        #
-        #   date   = (Time.now - 86400).httpdate
-        #   filter = "eventTimestamp ge #{date} and eventChannels eq 'Admin, Operation'"
-        #   select = "resourceGroupName, operationName"
-        #
-        #   ies.list_all(filter, select).each{ |event|
-        #     p event
-        #   }
-        #
-        def list_all(filter = nil, select = nil)
-          event_list = list(filter, select)
-          events = event_list.events
+          events.skip_token = get_skip_token(json_response['nextLink'])
 
-          while skip_token = event_list.skip_token do
-            event_list = list(filter, select, skip_token)
-            events += event_list.events
+          if options[:all]
```

Comment:
I couldn't get a simple recursive call to work without looping. Otherwise, you get an infinite loop.